serde-api/src/main/java/io/micronaut/serde/ObjectSerializer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.serde;
+
+import io.micronaut.core.annotation.Indexed;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.type.Argument;
+
+import java.io.IOException;
+
+/**
+ * A variation of {@link Serializer} that is serializing an object and supports serializing its content into an existing object.
+ *
+ * @param <T> The object type to be serialized
+ * @author Denis Stepanov
+ * @since 2.3
+ */
+@Indexed(ObjectSerializer.class)
+public interface ObjectSerializer<T> extends Serializer<T> {
```

Comment:
Perhaps this is a confusing name?

Reply:
What would you suggest? Before I had `SerializerInto` but I think it makes more sense to call it `ObjectSerializer` because `serializeInto` can only be used with a serializer representing JSON objects `{...}`.

Reply:
UnwrappedObjectSerializer maybe

Reply:
The idea is that an object serializer is any serializer representing an object, and if you put the unwrapped annotation, it will use `serializeInto` method to avoid creating an outer object.

Reply:
Going to merge it, and rename it later if you have a better suggestion